package/harvester-os/files/etc/systemd/system/rke2-shutdown.service

```diff
@@ -0,0 +1,14 @@
+[Unit]
+Description=Kill RKE2 Containers for clean unmount
+Documentation=man:systemd-halt.service(8)
+DefaultDependencies=no
+RefuseManualStart=yes
+Before=reboot.target halt.target shutdown.target poweroff.target umount.target final.target
+
+[Install]
+RequiredBy=reboot.target halt.target shutdown.target poweroff.target umount.target final.target
```

Comment:
Test and work well.
Is this unit required to be run **after** rke2-agent or rke2-server service is stopped?

Reply:
I feel like if it's a user that stopped/restarted the service, the user should decide whether `rke2-killall.sh` should be run.

Reply:
FWIW, I found some discussions about this issue, and seems like "stopping K3s/RKE2 does not stop running containers" is specifically designed to allow non-disruptive upgrade of K3s/RKE2.

- https://github.com/k3s-io/k3s/issues/2400#issuecomment-1040795816

Reply:
During an upgrade, we will always wait for RKE2 upgraded first and shut down later.